testsuite/MDAnalysisTests/coordinates/test_pdb.py

```diff
@@ -446,6 +454,61 @@ def test_stringio_outofrange(self, universe3):
             with mda.coordinates.PDB.PDBWriter(outstring) as writer:
                 writer.write(u.atoms)
 
+    def test_hetatm_written(self, universe4, tmpdir, outfile):
+        """
+        Checks that HETATM record types are written.
+        """
+
+        u = universe4
+        u_hetatms = u.select_atoms("resname ETA and record_type HETATM")
+        assert_equal(len(u_hetatms), 8)
+
+        u.atoms.write(outfile)
+        written = mda.Universe(outfile)
+        written_atoms = written.select_atoms("resname ETA and "
+                                             "record_type HETATM")
+
+        assert len(u_hetatms) == len(written_atoms), "mismatched HETATM number"
+        assert_almost_equal(u_hetatms.atoms.positions, written_atoms.atoms.positions)
+
+    def test_default_atom_record_type_written(self, universe5, tmpdir, outfile):
+        """
+        Checks that ATOM record types are written when there is no
+        record_type attribute.
+        """
+
+        u = universe5
+
+        expected_msg = ("Found no information for attr: "
+                        "'record_types' Using default value of 'ATOM'")
+
+        with pytest.warns(UserWarning, match=expected_msg):
+            u.atoms.write(outfile)
+
+        written = mda.Universe(outfile)
+        assert len(u.atoms) == len(written.atoms), "mismatched number of atoms"
+
+        atms = written.select_atoms("record_type ATOM")
+        assert len(atms.atoms) == len(u.atoms), "mismatched ATOM number"
+
+        hetatms = written.select_atoms("record_type HETATM")
+        assert len(hetatms.atoms) == 0, "mismatched HETATM number"
+
+    def test_abnormal_record_type(self, universe5, tmpdir, outfile):
+        """
+        Checks whether KeyError is raised when record type is
+        neither ATOM or HETATM.
+        """
+        u = universe5
+        u.add_TopologyAttr('record_type', ['ABNORM']*len(u.atoms))
+
+        expected_msg = ("Found 'ABNORM' for record type, but allowed "
+                        "types are ATOM or HETATM")
+
+        with pytest.raises(ValueError):
```

Comment:
This has changed now, you no longer raise the warning but pass the message to the ValueError. You need to match on the ValueError instead.

Reply:
Done, thanks for spotting this!